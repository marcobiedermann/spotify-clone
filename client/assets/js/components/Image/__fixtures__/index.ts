const image = {
  alt: '',
  height: 100,
  url: '',
  width: 100,
};

export default image;
