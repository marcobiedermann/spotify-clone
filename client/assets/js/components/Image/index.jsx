import PropTypes from 'prop-types';
import React from 'react';

const Image = ({
  alt,
  height,
  url,
  width,
}) => (
  <img
    alt={alt}
    height={height}
    src={url}
    width={width}
  />
);

Image.propTypes = {
  alt: PropTypes.string,
  height: PropTypes.number,
  url: PropTypes.string,
  width: PropTypes.number,
};

Image.defaultProps = {
  alt: '',
  height: 0,
  url: '',
  width: 0,
};

export default Image;
