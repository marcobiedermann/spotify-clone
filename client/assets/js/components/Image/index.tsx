import React, { FC } from 'react';

export interface ImageProps {
  alt: string;
  height?: number;
  url: string;
  width?: number;
}

const Image: FC<ImageProps> = (props) => {
  const { url, ...otherProps } = props;

  return <img src={url} {...otherProps} />;
};

export default Image;
