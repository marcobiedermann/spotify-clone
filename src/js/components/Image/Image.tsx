/* eslint-disable jsx-a11y/alt-text */

import React from 'react';

export interface ImageProps {
  alt: string;
  height: number;
  url: string;
  width: number;
}

function Image(props: ImageProps): JSX.Element {
  const { url, ...otherProps } = props;

  return <img src={url} {...otherProps} />;
}

export default Image;
