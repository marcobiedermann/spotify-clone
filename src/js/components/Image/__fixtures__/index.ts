import { ImageProps } from '..';

const image: ImageProps = {
  alt: 'Alt text',
  height: 300,
  url: 'https://i.scdn.co/image/ab67616d00001e02ff9ca10b55ce82ae553c8228',
  width: 300,
};

export default image;
