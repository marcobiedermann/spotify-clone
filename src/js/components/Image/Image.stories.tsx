import type { Meta, StoryObj } from '@storybook/react';
import Image from '.';
import imageFixtures from './__fixtures__';

const meta: Meta<typeof Image> = {
  component: Image,
  args: imageFixtures,
};

type Story = StoryObj<typeof Image>;

const Default: Story = {};

export { Default };
export default meta;
