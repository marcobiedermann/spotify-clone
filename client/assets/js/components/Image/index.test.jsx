import React from 'react';
import renderer from 'react-test-renderer';
import Image from '.';

it('renders correctly', () => {
  const tree = renderer
    .create(<Image />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
