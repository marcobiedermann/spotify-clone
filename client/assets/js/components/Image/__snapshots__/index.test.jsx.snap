// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt=""
  height={0}
  src=""
  width={0}
/>
`;
